### PR TITLE
Add enchantability parameter, resolve part of #39

### PR DESCRIPTION
**Pull request by Hubry**
#39 

Also fixes a few warnings and ~~moves libs to maven~~ removes IC2 that didn't work anyways. I hope you don't mind, and I hope I didn't mess up anything :P

### CODE REVIEW

File: src/main/java/xbony2/huesodewiki/infobox/InfoboxCreator.java [hunk 1]
```diff
@@ -23,7 +23,7 @@
 import xbony2.huesodewiki.infobox.parameters.*;
 
 public class InfoboxCreator {
-	public static List<IInfoboxParameter> parameters = new ArrayList();
+	public static List<IInfoboxParameter> parameters = new ArrayList<>();
```
Comment (xbony2) [inline src/main/java/xbony2/huesodewiki/infobox/InfoboxCreator.java:26]: ?

File: src/main/java/xbony2/huesodewiki/recipe/recipes/CraftingRecipe.java [hunk 2]
```diff
@@ -123,8 +123,8 @@ public String getRecipes(ItemStack itemstack){
 					ShapedOreRecipe shapedrecipe = (ShapedOreRecipe)recipe;
 					ret.append("{{Cg/Crafting Table\n");
 					
-					int maxHeight = shapedrecipe.getHeight();
-					int maxWidth = shapedrecipe.getWidth();
+					int maxHeight = shapedrecipe.getRecipeHeight();
+					int maxWidth = shapedrecipe.getRecipeWidth();
```
Comment (xbony2) [inline src/main/java/xbony2/huesodewiki/recipe/recipes/CraftingRecipe.java:127]: I assume this is removing a depreciation.

**Review by xbony2 — CHANGES_REQUESTED** [review]
no maven

File: src/main/java/xbony2/huesodewiki/recipe/recipes/CraftingRecipe.java (hunk 2 above)
Comment (Hubry) [reply src/main/java/xbony2/huesodewiki/recipe/recipes/CraftingRecipe.java:127]: Yes, that's exactly that. They call getHeight/getWidth anyways.

Comment (xbony2) [reply src/main/java/xbony2/huesodewiki/recipe/recipes/CraftingRecipe.java:127]: swell

File: src/main/java/xbony2/huesodewiki/infobox/InfoboxCreator.java (hunk 1 above)
Comment (Hubry) [reply src/main/java/xbony2/huesodewiki/infobox/InfoboxCreator.java:26]: My IDE complained about no <> there. Would show warnings too on a higher warning checking level I think, too. I think you have more of those but only fixed the one in the modified file.

Comment (xbony2) [reply src/main/java/xbony2/huesodewiki/infobox/InfoboxCreator.java:26]: @Hubry what's the exact error message?

Comment (xbony2) [reply src/main/java/xbony2/huesodewiki/infobox/InfoboxCreator.java:26]: err, warning message

Comment (Hubry) [reply src/main/java/xbony2/huesodewiki/infobox/InfoboxCreator.java:26]: "Unchecked assignment: `java.util.ArrayList` to `java.util.List<IInfoboxParameter>`."
So basically it wants a generic type and not a raw type.

Comment (xbony2) [reply src/main/java/xbony2/huesodewiki/infobox/InfoboxCreator.java:26]: Sounds like the compiler just being whiney. It's unneeded and ugly so ¯\\\_(ツ)\_/¯

Comment (elifoster) [reply src/main/java/xbony2/huesodewiki/infobox/InfoboxCreator.java:26]: It's a generics thing. You can read about it here: https://stackoverflow.com/questions/4166966/what-is-the-point-of-the-diamond-operator-in-java-7#4167148

**Pull request merged**